pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/BaseSingleSegmentConversionExecutor.java

```diff
@@ -132,10 +131,9 @@ public SegmentConversionResult executeTask(PinotTaskConfig pinotTaskConfig)
 
       // Collect the task processing metrics from various single segment executors and publish them here.
       SegmentMetadataImpl segmentMetadata = new SegmentMetadataImpl(indexDir);
-      Object numRecordsPurged = segmentConversionResult.getCustomProperty(PurgeTaskExecutor.NUM_RECORDS_PURGED_KEY);
+      Integer numRecordsPurged = segmentConversionResult.getNumRecordsPurged();
       if (numRecordsPurged != null) {
-        reportTaskProcessingMetrics(tableNameWithType, taskType, segmentMetadata.getTotalDocs(),
-            (int) numRecordsPurged);
+        reportTaskProcessingMetrics(tableNameWithType, taskType, segmentMetadata.getTotalDocs(), numRecordsPurged);
```

Comment:
@tibrewalpratik17 @Jackie-Jiang i re-used the existing `RECORDS_PURGED_COUNT` metric to track invalid records since it seems like it was meant to be somewhat generic. let me know if a dedicated metric is preferred instead.

Reply:
@dang-stripe can we create a new metric - `COMPACTED_RECORDS_COUNT` instead. It will help in clear isolation when PurgeTask and UpsertCompactionTask are enabled together on a table.